src/core/opamSystem.ml

```diff
@@ -821,13 +821,344 @@ let lock_none = {
 let lock_isatleast flag lock =
   lock_max flag lock.kind = lock.kind
 
+(* read_lines processes the lines of a text file in a platform-agnostic manner
+   (which is to say that it preserves '\r' on all platforms). It has two modes
+   of operation - in the Lines mode, it returns a string list of the lines, in
+   CrLf mode, it returns true if every line ends "\r\n". *)
+type _ read_return = Lines : string list read_return
+                   | CrLf  : bool read_return
```

Comment:
why not just define two functions ?

Reply:
Where's the fun in that 🙂 More seriously, they're very similar, but not quite identical functions

Reply:
i'd argue that you could factorise the core in `read_lines_aux`, and define the two functions with just the last lines. But well, doesn't really matter ^^.